docs/userguide/networking/dockernetworks.md

```diff
@@ -0,0 +1,368 @@
+<!--[metadata]>
++++
+title = "Docker container networking"
+description = "How do we connect docker containers within and across hosts ?"
+keywords = ["Examples, Usage, network, docker, documentation, user guide, multihost, cluster"]
+[menu.main]
+parent = "smn_networking"
+weight = -5
++++
+<![end-metadata]-->
+
+# Understand Docker container networks
+
+To build web applications that act in concert but do so securely. Networks by definition provide complete isolation for containers. So, it is important to have control over the networks your applications run on. Docker container networks give you that control.
+
+This section provides an overview of the default networking behavior that Docker Engine delivers natively.  It describes the resources required to create networks on a single host or across a cluster of hosts.
+
+## Default Networks
+
+When you install Docker, it creates three networks automatically. You can list these networks using the `docker network ls` command:
+
+```
+$ docker network ls
+NETWORK ID          NAME                DRIVER
+7fca4eb8c647        bridge              bridge
+9f904ee27bf5        none                null
+cf03ee007fb4        host                host
+```
+
+Historically, these three networks are  part of Docker's implementation. When you run a container you can use the `--net` flag to specify which network you want to run a container on. These three networks are still available to you.
+
+The `bridge` network represents the `docker0` network present in all Docker installations. The Docker daemon launches each new container this network by default. You can see this bridge as part of the hosts network stack by doing an `ifconfig` on the host.
+
+```
+ubuntu@ip-172-31-36-118:~$ ifconfig
+docker0   Link encap:Ethernet  HWaddr 02:42:47:bc:3a:eb  
+          inet addr:172.17.0.1  Bcast:0.0.0.0  Mask:255.255.0.0
+          inet6 addr: fe80::42:47ff:febc:3aeb/64 Scope:Link
+          UP BROADCAST RUNNING MULTICAST  MTU:9001  Metric:1
+          RX packets:17 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:0
+          RX bytes:1100 (1.1 KB)  TX bytes:648 (648.0 B)
+...
+
+The `none` network adds a container to a container-specific network stack.  That container lacks a network interface. Attaching to such a container and looking at it's stack you see this:
+
+```
+ubuntu@ip-172-31-36-118:~$ docker attach nonenetcontainer
+
+/ # cat /etc/hosts
```

Comment:
Although probably the actual output. Could you change the prompt in this example to our standard format? (just `$`, no leading `/`)